contracts/KyberOasisReserve.sol

```diff
@@ -0,0 +1,273 @@
+pragma solidity 0.4.18;
+
+import "./ERC20Interface.sol";
+import "./Utils2.sol";
+import "./Withdrawable.sol";
+import "./KyberReserveInterface.sol";
+
+
+contract OtcInterface {
+    function getBuyAmount(ERC20 buyGem, ERC20 payGem, uint payAmt) public constant returns (uint fillAmt);
+    function sellAllAmount(ERC20 payGem, uint payAmt, ERC20 buyGem, uint minFillAmount) public returns (uint fillAmt);
+}
+
+
+contract TokenInterface is ERC20 {
+    function deposit() public payable;
+    function withdraw(uint) public;
+}
+
+
+contract KyberOasisReserve is KyberReserveInterface, Withdrawable, Utils2 {
+
+    uint constant internal ETH_SRC_AMOUNT = (10**18);
+    address public kyberNetwork;
+    OtcInterface public otc;
+    TokenInterface public wethToken;
+    ERC20 public tradeToken;
+    bool public tradeEnabled;
+    uint public feeBps;
+
+    function KyberOasisReserve(
+        address _kyberNetwork,
+        OtcInterface _otc,
+        TokenInterface _wethToken,
+        ERC20 _tradeToken,
+        address _admin,
+        uint _feeBps
+    ) public {
+        require(_admin != address(0));
+        require(_kyberNetwork != address(0));
+        require(_otc != address(0));
+        require(_wethToken != address(0));
+        require(_tradeToken != address(0));
+        require(_feeBps < 10000);
+        require(getDecimals(_wethToken) == MAX_DECIMALS);
+        require(getDecimals(_tradeToken) == MAX_DECIMALS);
+
+        kyberNetwork = _kyberNetwork;
+        otc = _otc;
+        wethToken = _wethToken;
+        tradeToken = _tradeToken;
+        admin = _admin;
+        feeBps = _feeBps;
+        tradeEnabled = true;
+
+        wethToken.approve(otc, 2**255);
+        tradeToken.approve(otc, 2**255);
+    }
+
+    event DepositToken(ERC20 token, uint amount);
+
+    /* solhint-disable no-complex-fallback */
+    function() public payable {
+        require(msg.sender == address(wethToken));
+        DepositToken(ETH_TOKEN_ADDRESS, msg.value);
+    }
+    /* solhint-enable no-complex-fallback */
+
+    event TradeExecute(
+        address indexed origin,
```

Comment:
`tx.origin` has different meaning in solidity. why not call it `sender`? it is `msg.sender`, not `tx.origin`.

Reply:
I copied it from the regular reserve. Let's leave it identical.